plugins/webhooks/index.js

```diff
@@ -14,25 +14,49 @@ const CHECKOUT_URL_SCHEMA = schema.config.regex.CHECKOUT_URL;
 const CHECKOUT_URL_SCHEMA_REGEXP = new RegExp(CHECKOUT_URL_SCHEMA);
 const DEFAULT_MAX_BYTES = 1048576;
 
+/**
+ * Check if the tag or release filtering or not
+ * @param {String}    action          SCM webhook action type
+ * @param {Array}     workflowGraph   pipeline workflowGraph
+ * @returns {Boolean} isFilterlingEnabled
+ */
+function isReleaseOrTagFilteringEnabled(action, workflowGraph) {
+    let isFilterlingEnabled = false;
+
+    workflowGraph.edges.forEach(edge => {
+        const releaseOrTagRegExp = action === 'release' ? new RegExp('^~(release):') : new RegExp('^~(tag):');
+
+        if (releaseOrTagRegExp) {
+            if (edge.src.match(releaseOrTagRegExp)) {
+                isFilterlingEnabled = true;
+            }
+        }
+    });
+
+    return isFilterlingEnabled;
+}
 /**
  * Determine "startFrom" with type, action and branches
- * @param {String} action          SCM webhook action type
- * @param {String} type            Triggered SCM event type ('pr' or 'repo')
- * @param {String} targetBranch    The branch against which commit is pushed
- * @param {String} pipelineBranch  The pipeline branch
- * @returns {String}               startFrom
+ * @param {String}   action                    SCM webhook action type
+ * @param {String}   type                      Triggered SCM event type ('pr' or 'repo')
+ * @param {String}   targetBranch              The branch against which commit is pushed
+ * @param {String}   pipelineBranch            The pipeline branch
+ * @param {String}   releaseName               SCM webhook release name
+ * @param {String}   tagName                   SCM webhook tag name
+ * @param {Boolean}  isReleaseOrTagFiltering   Check if the tag or release filtering or not
+ * @returns {String} startFrom
  */
-function determineStartFrom(action, type, targetBranch, pipelineBranch) {
+function determineStartFrom(action, type, targetBranch, pipelineBranch, releaseName, tagName, isReleaseOrTagFiltering) {
     let startFrom;
 
     if (type && type === 'pr') {
         startFrom = '~pr';
     } else {
         switch (action) {
             case 'release':
-                return '~release';
+                return isReleaseOrTagFiltering && action === 'release' ? `~release:${releaseName}` : '~release';
```

Comment:
I think it is better to check if `releaseName` and `tagName` are undefined.

Reply:
Add undefined check.